index.js

```diff
@@ -107,6 +107,13 @@ function serveIndex(root, options) {
       return;
     }
 
+    // content-negotiation
+    var accept = accepts(req);
+    var type = accept.type(mediaTypes);
+
+    // not acceptable
+    if (!type) return next(createError(406));
```

Comment:
Makes sense.

I originally moved it up so that it would be checked before all of the fs.stat()s. I moved it just after the directory stat and before the fs.readdir.